src/kili/services/copy_project/__init__.py

```diff
@@ -0,0 +1,139 @@
+"""
+Copy project implementation.
+"""
+from typing import Dict, Optional
+
+from kili.authentication import KiliAuth
+from kili.mutations.project import MutationsProject
+from kili.queries.project import QueriesProject
+from kili.queries.project_user import QueriesProjectUser
+
+
+class CopyProject:  # pylint: disable=too-few-public-methods
+    """
+    Class for copying an existing project.
+    """
+
+    FIELDS_PROJECT = ["title", "jsonInterface", "inputType", "description", "id"]
+    FIELDS_QUALITY_SETTINGS = [
+        "consensusTotCoverage",
+        "consensusMark",
+        "minConsensusSize",
+        "honeypotMark",
+        "useHoneyPot",
+        "reviewCoverage",
+    ]
+
+    def __init__(self, auth: KiliAuth) -> None:
+        self.auth = auth
+
+        mutations_project = MutationsProject(self.auth)
+        queries_project = QueriesProject(self.auth)
+        queries_project_user = QueriesProjectUser(self.auth)
+
+        self.projects = queries_project.projects
+        self.create_project = mutations_project.create_project
+        self.append_to_roles = mutations_project.append_to_roles
+        self.update_properties_in_project = mutations_project.update_properties_in_project
+        self.project_users = queries_project_user.project_users
+
+    def copy_project(  # pylint: disable=too-many-arguments
+        self,
+        from_project_id: str,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+        copy_json_interface: bool = True,
+        copy_quality_settings: bool = True,
+        copy_members: bool = True,
+    ) -> str:
+        """Copy an existing project.
+
+        Copy an existing source project from its ID.
+
+        Args:
+            from_project_id: Project ID to copy from.
+            title: Title for the new project. Defaults to source project
+                title if `None` is provided.
+            description: Description for the new project. Defaults to empty string
+                if `None` is provided.
+            copy_json_interface: Copy the json interface from the source project to the new one.
+            copy_quality_settings: Copy the quality settings from the source project to the new one.
+            copy_members: Copy the members from the source project to the new one.
+
+        Returns:
+            The created project ID.
+
+        Examples:
+            >>> kili.copy_project(from_project_id="clbqn56b331234567890l41c0")
+        """
+        fields = self.FIELDS_PROJECT
+        if copy_quality_settings:
+            fields = fields + self.FIELDS_QUALITY_SETTINGS
+
+        src_project = self.projects(  # type: ignore
+            project_id=from_project_id,
+            fields=fields,
+        )[0]
```

Comment:
before to get the first element of the list, assert that you got one project, otherwise return an error